### PR TITLE
Updating libindy to v1.4.0

### PR DESCRIPTION
**Pull request by pimotte**
Decided to not update claim->credential terminology in backends for now.

### CODE REVIEW

File: indy-wrapper/docker-compose.yml [hunk 1]
```diff
@@ -4,6 +4,8 @@ services:
     build:
       context: .
     network_mode: "host"
+    environment:
```
Comment (PJUllrich) [inline indy-wrapper/docker-compose.yml]: Delete this line in order to not have a super spammy build log

File: indy-wrapper/src/main/java/nl/quintor/studybits/indy/wrapper/Issuer.java [hunk 2]
```diff
@@ -2,114 +2,99 @@
...
-                })).thenApply((response) -> {
-                            log.debug("{} Got ClaimDefTxn response: {}", name, response);
-                            return response;
-                        }
-                );
+                {
+                    return Anoncreds.issuerCreateAndStoreCredentialDef(wallet.getWallet(), getIssuerDid(), schema.toJSON(), name + "_" + schemaId, "CL", "{\"support_revocation\":false}");
+                }))
+                .thenCompose(wrapException(createAndStoreCredentialDefResult -> {
+                    return Ledger.buildCredDefRequest(getIssuerDid(), createAndStoreCredentialDefResult.getCredDefJson())
```
Comment (PJUllrich) [inline indy-wrapper/src/main/java/nl/quintor/studybits/indy/wrapper/Issuer.java]: add debug loggers here. We always have problems getting information about what goes wrong in the futures here, so debug loggers are crucial!

Comment (PJUllrich) [reply indy-wrapper/src/main/java/nl/quintor/studybits/indy/wrapper/Issuer.java]: In particular, I mean add a debug log for the input parameter 'createAndStoreCredentialDefResult'

File: indy-wrapper/src/main/java/nl/quintor/studybits/indy/wrapper/dto/CredentialInfo.java [hunk 3]
```diff
@@ -7,16 +7,24 @@
...
-    @JsonProperty( "schema_key" )
-    private SchemaKey schemaKey;
-    @JsonProperty( "issuer_did" )
-    private String issuerDid;
-    @JsonProperty( "revoc_reg_seq_no" )
-    private Integer revocRegSeqNo;
-}
+
+    @JsonProperty( "rev_reg_id" )
+    private Integer revRegId;
```
Comment (PJUllrich) [inline indy-wrapper/src/main/java/nl/quintor/studybits/indy/wrapper/dto/CredentialInfo.java]: What is `revRegId`? I assume `credRevId` is `Credential Reference Id`? These variable names are not too informative. Consider renaming.

File: indy-wrapper/src/main/java/nl/quintor/studybits/indy/wrapper/dto/CredentialOffer.java [hunk 4]
```diff
@@ -11,11 +11,13 @@
...
-public class ClaimOffer implements Serializable, AuthCryptable {
+public class CredentialOffer implements Serializable, AuthCryptable {
     @JsonProperty("issuer_did")
     private String issuerDid;
-    @JsonProperty("schema_key")
-    private SchemaKey schemaKey;
+    @JsonProperty("schema_id")
+    private String schemaId;
+    @JsonProperty("cred_def_id")
+    private String credDefId;
```
Comment (PJUllrich) [inline indy-wrapper/src/main/java/nl/quintor/studybits/indy/wrapper/dto/CredentialOffer.java]: Maybe rename the cryptic `credDefId` to `definitionId`

File: student/src/main/java/nl/quintor/studybits/student/entities/ClaimEntity.java [hunk 5]
```diff
@@ -43,13 +42,17 @@
...
-    private String issuerDid;
-
-    private String myDid;
-
-    @Column(nullable = false)
-    private String theirDid;
-
-
+    @Override
+    public String toString() {
```
Comment (PJUllrich) [inline student/src/main/java/nl/quintor/studybits/student/entities/ClaimEntity.java]: Why are you creating your own JSON string here? Can't you use Dozer for custom mapping? 
Also, don't send the Student object, but only the student id.

File: student/src/main/java/nl/quintor/studybits/student/entities/Student.java [hunk 6]
```diff
@@ -36,4 +36,17 @@
 
     @OneToOne
     private MetaWallet metaWallet;
+
+    @Override
+    public String toString() {
```
Comment (PJUllrich) [inline student/src/main/java/nl/quintor/studybits/student/entities/Student.java]: Same holds here. Why are you creating your own JSON string?

File: student/src/main/resources/mappings/ClaimModel.xml [hunk 7]
```diff
@@ -16,8 +16,8 @@
             <b>issuingUniversityName</b>
         </field>
         <field type="one-way">
```
Comment (PJUllrich) [inline student/src/main/resources/mappings/ClaimModel.xml]: Delete the "one-way" here

File: university/src/main/java/nl/quintor/studybits/university/controllers/admin/SchemaController.java [hunk 8]
```diff
@@ -17,22 +19,19 @@
...
-        return mapper.map(record, SchemaDefinitionModel.class);
-    }
-
     @GetMapping
     List<SchemaDefinitionModel> getSchemaDefinitions() {
         return universityService
                 .getSchemaDefinitions(userContext.currentUniversityName())
                 .stream()
-                .map(this::toModel)
+                .distinct()
```
Comment (PJUllrich) [inline university/src/main/java/nl/quintor/studybits/university/controllers/admin/SchemaController.java]: Is there no need for mapping the entity to a model anymore?

File: university/src/main/java/nl/quintor/studybits/university/entities/ClaimSchema.java [hunk 9]
```diff
@@ -47,11 +50,25 @@
...
+        this.schemaId = schemaId;
         this.university = university;
         this.schemaName = schemaName;
         this.schemaVersion = schemaVersion;
         this.schemaIssuerDid = schemaIssuerDid;
-        this.claimDefined = false;
+    }
+
+    @Override
+    public String toString() {
```
Comment (PJUllrich) [inline university/src/main/java/nl/quintor/studybits/university/entities/ClaimSchema.java]: See comments about creating your own JSON strings from above

File: student/src/main/java/nl/quintor/studybits/student/entities/ClaimEntity.java (hunk 5 above)
Comment (pimotte) [reply student/src/main/java/nl/quintor/studybits/student/entities/ClaimEntity.java]: This is not JSON, just a readable toString for debug logging

File: university/src/main/java/nl/quintor/studybits/university/controllers/admin/SchemaController.java (hunk 8 above)
Comment (pimotte) [reply university/src/main/java/nl/quintor/studybits/university/controllers/admin/SchemaController.java]: Nope, as the getSchemaDefintions now returns models

**Pull request merged**